code/controllers/subsystem/movement/movement_types.dm

```diff
@@ -0,0 +1,871 @@
+ * skip_first -  Whether or not to delete the first item in the path. This would be done because the first item is the starting tile, which can break things
+ * timeout - Time in deci-seconds until the moveloop self expires. Defaults to infinity
+ * subsystem - The movement subsystem to use. Defaults to SSmovement. Only one loop can exist for any one subsystem
+ * priority - Defines how different move loops override each other. Lower numbers beat higher numbers, equal defaults to what currently exists. Defaults to MOVEMENT_DEFAULT_PRIORITY
+ * flags - Set of bitflags that effect move loop behavior in some way. Check _DEFINES/movement.dm
+ *
+**/
+/datum/controller/subsystem/move_manager/proc/hostile_jps_move(moving,
+	chasing,
+	delay,
+	timeout,
+	repath_delay,
+	max_path_length,
+	minimum_distance,
+	obj/item/card/id/id,
+	simulated_only,
+	turf/avoid,
+	skip_first,
+	subsystem,
+	priority,
+	flags,
+	datum/extra_info)
+	return add_to_loop(moving,
+		subsystem,
+		/datum/move_loop/has_target/jps/hostile,
+		priority,
+		flags,
+		extra_info,
+		delay,
+		timeout,
+		chasing,
+		repath_delay,
+		max_path_length,
+		minimum_distance,
+		id,
+		simulated_only,
+		avoid,
+		skip_first)
+
+/datum/move_loop/has_target/jps/hostile
+	var/target_turf
+
+/datum/move_loop/has_target/jps/hostile/recalculate_path()
+	. = ..()
+	// Implementing pathfinding fallback solution
+	if(!movement_path)
+		var/ln = get_dist(moving, target)
+		var/turf/target_new = target
+		var/found_blocker
+		while(!movement_path && (ln > 0)) //will stop if we can find a valid path or if ln gets reduced to 0 or less
+			find_target:
+				for(var/i in 1 to ln) //calling get_path_to every time is quite taxing lets see if we can find whatever blocks us
+					target_new = get_step(target_new,  get_dir(target_new, moving)) //step towards the origin until we find the blocker then 1 further
+					ln--
+					if(target_new.density && !(target_new.pass_flags_self & moving.pass_flags)) //we check for possible tiles that could block us
+						found_blocker = TRUE
+						continue find_target //in case there is like a double wall
+					for(var/obj/o in target_new.contents)
+						if(o.density && !(o.pass_flags_self & moving.pass_flags)) //We check for possible blockers on the tile
+							found_blocker = TRUE
+							continue find_target
+					if(found_blocker) //cursed but after we found the blocker we end the loop on the next illiteration
+						break find_target
+			found_blocker = FALSE
+			movement_path = get_path_to(moving, target_new, max_path_length, 0, id, simulated_only, avoid, skip_first) //here the min distance is always 0 because we need to stand beside the blocker
+	target_turf = get_turf(target)
+
+/datum/move_loop/has_target/jps/hostile/move()
+	var/atom/movable/atom = moving
+	if(!length(movement_path))
+		INVOKE_ASYNC(src, .proc/recalculate_path)
+		if(!length(movement_path))
+			return FALSE
```

Comment:
Well its true that those two checks can be intigrated with each other easily i sadly seem to have overlooked this and forgot to rewrite it later on for the check after the move proc it is kinda needed cause it is supposed to ensure that we actually where able to step towards the target turf if not it will recalculate the path cause that would indicate something pushed the atom far enough so it was unable to reach its true destination. However maybe i should change this check up a little so it only fires if the destination turf is also not in range of 1 cause if that is the case we can just retry the next time and do not have to recalculate this could mainly happen for bumping into an airlock as an example.